tests/net/tcp_socket.cpp

```diff
@@ -0,0 +1,337 @@
+#include <gtest/gtest.h>
+#include <net/tcp_socket.hpp>
+#include <net/exception.hpp>
+#include <future>
+#include <string>
+#include <map>
+#include <sstream>
+
+///*
+TEST(TCPSocket, SendAndReceive) {
```

Comment:
This is not a unit test and it shouldn't probably be part of standard test suite. We will have to think about separating component tests from unit tests.